discord/onboarding.py

```diff
@@ -0,0 +1,319 @@
+        self.id: int | None = id or randint(10000000000000000)
+        self.type: PromptType = type
+        if isinstance(self.type, int):
+            self.type = try_enum(PromptType, self.type)
+        self.options: list[PromptOption] = options
+        self.title: str = title
+        self.single_select: bool = single_select
+        self.required: bool = required
+        self.in_onboarding: bool = in_onboarding
+
+    def __repr__(self):
+        return f"<OnboardingPrompt (title={self.title} required={self.required})>"
+
+    def to_dict(self) -> OnboardingPromptPayload:
+        dict_: OnboardingPromptPayload = {
+            "type": self.type.value,
+            "title": self.title,
+            "single_select": self.single_select,
+            "required": self.required,
+            "in_onboarding": self.in_onboarding,
+            "options": [option.to_dict() for option in self.options],
+            "id": self.id,
+        }
+
+        return dict_
+
+    @classmethod
+    def _from_dict(
+        cls, data: OnboardingPromptPayload, guild: Guild
+    ) -> OnboardingPrompt:
+        id = data.get("id")
+        type = try_enum(PromptType, data.get("type"))
+        title = data.get("title")
+        single_select = data.get("single_select")
+        required = data.get("required")
+        in_onboarding = data.get("in_onboarding")
+        options = [
+            PromptOption._from_dict(option, guild) for option in data.get("options", [])
+        ]
+
+        return cls(type=type, title=title, single_select=single_select, required=required, in_onboarding=in_onboarding, options=options, id=id)  # type: ignore
+
+
+class Onboarding:
+    """Represents the Onboarding flow for a guild.
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+
+    prompts: List[:class:`OnboardingPrompt`]
+        A list of prompts displayed in the onboarding flow.
+    enabled: :class:`bool`
+        Whether onboarding is enabled in the guild.
+    mode: :class:`OnboardingMode`
+        The current onboarding mode.
+    """
+
+    def __init__(self, data: OnboardingPayload, guild: Guild):
+        self._guild = guild
+        self._update(data)
+
+    def __repr__(self):
+        return f"<Onboarding enabled={self.enabled} default_channels={self.default_channels_channels}>"
+
+    def _update(self, data: OnboardingPayload):
+        self.guild_id: Snowflake = data["guild_id"]
+        self.prompts: list[OnboardingPrompt] = [
+            OnboardingPrompt._from_dict(prompt, self._guild)
+            for prompt in data.get("prompts", [])
+        ]
+        self.default_channel_ids: list[int] = [
+            int(c) for c in data["default_channel_ids"]
+        ]
+        self.enabled: bool = data["enabled"]
+        self.mode: OnboardingMode = try_enum(OnboardingMode, data.get("mode"))
+
+    @property
+    def guild(self) -> Guild:
+        """The guild this onboarding flow belongs to."""
+        return self._guild

```

Comment:
This suggestion replaces the `_guild` attribute and the `guild` property with a `guild` attribute
```suggestion
        self.guild = guild
        self._update(data)

    def __repr__(self):
        return f"<Onboarding enabled={self.enabled} default_channels={self.default_channels_channels}>"

    def _update(self, data: OnboardingPayload):
        self.guild_id: Snowflake = data["guild_id"]
        self.prompts: list[OnboardingPrompt] = [
            OnboardingPrompt._from_dict(prompt, self.guild)
            for prompt in data.get("prompts", [])
        ]
        self.default_channel_ids: list[int] = [
            int(c) for c in data["default_channel_ids"]
        ]
        self.enabled: bool = data["enabled"]
        self.mode: OnboardingMode = try_enum(OnboardingMode, data.get("mode"))